src/Emit.ts

```diff
@@ -0,0 +1,65 @@
+import * as fs from "fs";
+import * as path from "path";
+import * as ts from "typescript";
+import { CompilerOptions, LuaLibImportKind } from "./CompilerOptions";
+import { TranspiledFile } from "./Transpile";
+
+const trimExt = (filePath: string) =>
+    path.join(path.dirname(filePath), path.basename(filePath, path.extname(filePath)));
+
+let lualibContent: string;
+export function emitTranspiledFiles(
```

Comment:
Like discussed on discord too, I would initially say this functionality should belong with the CLI (only).